src/context.rs

```diff
@@ -404,3 +408,123 @@ impl<A> FusedFuture for ReceiveFuture<A> {
         self.0.is_terminated()
     }
 }
+
+pub struct TickFuture<'a, A> {
+    state: TickState<'a, A>,
+    span: HandlerSpan,
+}
+
+impl<'a, A> TickFuture<'a, A> {
+    /// Return the handler's [`tracing::Span`](https://docs.rs/tracing/latest/tracing/struct.Span.html),
+    /// creating it if it has not already been created. This can be used to log messages into the
+    /// span when required, such as if it is cancelled later due to a timeout.
+    ///
+    /// ```rust
+    /// # use std::ops::ControlFlow;
+    /// # use std::time::Duration;
+    /// # use tokio::time::timeout;
+    /// # use xtra::prelude::*;
+    /// #
+    /// # struct MyActor;
+    /// # #[async_trait::async_trait] impl Actor for MyActor { type Stop = (); async fn stopped(self) {} }
+    /// #
+    /// # let mut actor = MyActor;
+    /// # let (addr, mut ctx) = Context::<MyActor>::new(None);
+    /// # drop(addr);
+    /// # tokio::runtime::Runtime::new().unwrap().block_on(async {
+    /// # actor.started(&mut ctx).await;
+    /// #
+    /// # loop {
+    /// # let msg = ctx.next_message().await;
+    ///  let mut fut = ctx.tick(msg, &mut actor);
+    ///  let span = fut.get_or_create_span().clone();
+    ///  match timeout(Duration::from_secs(1), fut).await {
+    ///      Ok(ControlFlow::Continue(())) => (),
+    ///      Ok(ControlFlow::Break(())) => break actor.stopped().await,
+    ///      Err(_elapsed) => {
+    ///          let _entered = span.enter();
+    ///          span.record("interrupted", &"timed_out");
+    ///          tracing::warn!(timeout_seconds = 1, "Handler execution timed out");
+    ///      }
+    ///  }
+    /// # } })
+    /// ```
+    ///
+    #[cfg(feature = "instrumentation")]
+    pub fn get_or_create_span(&mut self) -> &tracing::Span {
```

Comment:
Would it be more ergonomic to not return a reference here?

Reply:
We are immediately cloning it in the example and the reference keeps the mutable borrow alive which means the future cannot be polled during that time. I don't see how a user would not immediately clone the span to get rid of the reference.

Reply:
> I don't see how a user would not immediately clone the span to get rid of the reference.

There are some functions that you can use on the span which only take &self, such as [`record`](https://docs.rs/tracing/latest/tracing/span/struct.Span.html#method.record), which is why I kept it like this with an explicit clone in most cases

Reply:
Fair enough.